group_project/lib/screens/sign_up_screen.dart

```diff
@@ -0,0 +1,214 @@
+import 'package:flutter/material.dart';
+import '../services/auth_service.dart';
+import 'verify_email_screen.dart';
+
+class SignUpScreen extends StatefulWidget {
+  const SignUpScreen({super.key});
+
+  @override
+  State<SignUpScreen> createState() => _SignUpScreenState();
+}
+
+class _SignUpScreenState extends State<SignUpScreen> {
+  final _formKey = GlobalKey<FormState>();
+  final _emailCtrl = TextEditingController();
+  final _pwdCtrl = TextEditingController();
+
+  bool _obscure = true;
+  bool _agreed = false;
+  bool _loading = false;
+
+  @override
+  void dispose() {
+    _emailCtrl.dispose();
+    _pwdCtrl.dispose();
+    super.dispose();
+  }
+
+  String? _validateEmail(String? v) {
+    if (v == null || v.trim().isEmpty) return 'Email is required';
+    final re = RegExp(r'^[^@\s]+@[^@\s]+\.[^@\s]+$');
+    if (!re.hasMatch(v.trim())) return 'Enter a valid email';
+    return null;
+  }
+
+  String? _validatePassword(String? v) {
+    if (v == null || v.isEmpty) return 'Password is required';
+    if (v.length < 6) return 'At least 6 characters';
+    return null;
+  }
+
+  Future<void> _submit() async {
+    debugPrint('👉 _submit called');
+    final ok = _formKey.currentState?.validate() ?? false;
+    if (!ok) return;
+    if (!_agreed) {
+      debugPrint('❌ Terms not agreed');
+      ScaffoldMessenger.of(context).showSnackBar(
+        const SnackBar(content: Text('Please accept Terms and Privacy Policy')),
+      );
+      return;
+    }
+
+    setState(() => _loading = true);
+    try {
+      final _authService = AuthService();
+      final email = _emailCtrl.text.trim();
+      final pwd = _pwdCtrl.text;
+
+      // Step 1: Create account
+      debugPrint('👉 Creating account for $email');
+      await AuthService.instance.signUpWithEmail(email, pwd);
+
+      // Step 2: Send verification email
+      debugPrint('👉 Sending verification email...');
+      await AuthService.instance.sendEmailVerification();
```

Comment:
```suggestion
      await AuthService.sendEmailVerification();
```
Since you use a singleton pattern for the authservice, there's no need to call it as an instance throughout the code.